TestRangeCalculator.cpp

```diff
@@ -0,0 +1,18 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "catch.hpp"
+#include "RangeCalculator.h"
+
+//Test Driven Ranges
+
+//passing test
+TEST_CASE("Case for detecting number of continuous range in the reading array given") {
+  int ChargingCurrentReadings[] = {3, 3, 5, 4, 10, 11, 12};
+  int numOfCCReadings = sizeof(ChargingCurrentReadings) / sizeof(ChargingCurrentReadings[0]);
+  REQUIRE(NumberofContinuousRange(ChargingCurrentReadings, numOfCCReadings) == 2);
+  
+  //Failing test
+  TEST_CASE("Case for detecting number of continuous range in the reading array given") {
+  int ChargingCurrentReadings[] = {3, 3, 5, 4};
+  int numOfCCReadings = sizeof(ChargingCurrentReadings) / sizeof(ChargingCurrentReadings[0]);
+  REQUIRE(NumberofContinuousRange(ChargingCurrentReadings, numOfCCReadings) == 2);
```

Comment:
It looks like this test fails because the test is wrong (expects two ranges when there is only one?)
Keep the tests as statements of functionality. Let the tests be correct, but fail because implementation is not there. E.g., pass null for the ChargingCurrentReadings and expect it to return 0. It will fail. Then implement to pass.
Implement new stuff in new functions, avoid fitting all functionality into the same function.